compendium/DeclarativeServices/test/TestConfigPolicies.cpp

```diff
@@ -0,0 +1,340 @@
+    auto instanceI = GetInstance<test::CAInterface>();
+    ASSERT_TRUE(instanceI) << "GetService failed for CAInterface";
+
+    //Confirm component instance was created with the correct properties
+    instanceProps = instanceI->GetProperties();
+    auto uniquePropI = instanceProps.find(uniqueProp[i]);
+
+    EXPECT_TRUE(uniquePropI != instanceProps.end())
+      << uniqueProp[i] << " not found in constructed instance"; //not found
+    EXPECT_EQ(uniquePropI->second, instance[i]);
+
+    ++i;
+  }
+}
+
+TEST_F(tServiceComponent, testRequiredConfigPolicy) // DS_CAI_FTC_6
+{
+  // Start the test bundle containing the component name.
+  std::string componentName           = "sample::ServiceComponentCA20";
+  cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSCA20");
+
+  // Use DS runtime service to validate the component state
+  scr::dto::ComponentDescriptionDTO compDescDTO;
+  auto compConfigs =
+    GetComponentConfigs(testBundle, componentName, compDescDTO);
+  EXPECT_EQ(compConfigs.at(0).state,
+            scr::dto::ComponentState::UNSATISFIED_REFERENCE)
+    << "UNSATISFIED_REFERENCE is expected because configuration object is not "
+       "created.";
+
+  // Get a service reference to ConfigAdmin to create the component instance.
+  auto configAdminService =
+    GetInstance<cppmicroservices::service::cm::ConfigurationAdmin>();
+  ASSERT_TRUE(configAdminService) << "GetService failed for ConfigurationAdmin";
+
+  // Create configuration object and update property
+  auto configuration  = configAdminService->GetConfiguration(componentName);
+  auto configInstance = configuration->GetPid();
+
+  cppmicroservices::AnyMap props(
+    cppmicroservices::AnyMap::UNORDERED_MAP_CASEINSENSITIVE_KEYS);
+  const std::string instanceId{ "instance1" };
+  props["uniqueProp"] = instanceId;
+  configuration->Update(props);
+
+  // The Update properties sends an asynchronous request to DS to
+  // update the properties.
+  auto result = RepeatTaskUntilOrTimeout(
+    [&compDescDTO, &compConfigs, this, &testBundle, &configInstance]() {
+      compDescDTO = dsRuntimeService->GetComponentDescriptionDTO(
+        testBundle, configInstance);
+      if (compDescDTO.name != "") {
+        compConfigs =
+          this->dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+      }
+    },
+    [&compConfigs, &instanceId]() -> bool {
+      if (compConfigs.size() == 1) {
+        auto properties = compConfigs.at(0).properties;
+        auto id = properties.find("uniqueProp");
+        if (id != properties.end()) {
+          return (id->second == instanceId);
+        }
+      }
+      return false;
+    });
+
+  ASSERT_TRUE(result) << "Timed out waiting for Update Configuration"
+                         "to complete.";
+
+  // Confirm configuration object presented and component is satisfied.
+  compConfigs = dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
```

Comment:
Already did the GetComponentConfigurationDTOs in the last block of code. Don't need to do it again? Just check the state?